bindings/core.ts

```diff
@@ -0,0 +1,130 @@
+import { bindSolcMethod, bindSolcMethodWithFallbackFunc } from './helpers';
+import translate from '../translate';
+import * as semver from 'semver';
+
+export function setupCore (solJson) {
+  const core = {
+    alloc: bindAlloc(solJson),
+    license: bindLicense(solJson),
+    version: bindVersion(solJson),
+    reset: bindReset(solJson)
+  };
+
+  const helpers = {
+    addFunction: bindAddFunction.bind(this, solJson),
+    removeFunction: bindRemoveFunction.bind(this, solJson),
+
+    copyFromCString: bindCopyFromCString.bind(this, solJson),
+    copyToCString: bindCopyToCString.bind(this, solJson, core.alloc),
+
+    // @ts-ignore
+    versionToSemver: versionToSemver(core.version())
+  };
+
+  return {
+    ...core,
+    ...helpers,
+
+    isVersion6OrNewer: semver.gt(helpers.versionToSemver(), '0.5.99')
+  };
+}
+
+/**********************
+ * Core Functions
+ **********************/
+
+/**
+ * Returns a binding to the solidity_alloc function.
+ *
+ * @param solJson
+ */
+function bindAlloc (solJson) {
+  return bindSolcMethodWithFallbackFunc(solJson,
+    'solidity_alloc', 'number', ['number'], 'malloc');
+}
+
+/**
+ * Returns a binding to the solidity_version method.
+ *
+ * @param solJson
+ */
+function bindVersion (solJson) {
+  return bindSolcMethodWithFallbackFunc(solJson, 'solidity_version', 'string',
+    ['number'], 'version', null);
+}
```

Comment:
Yeah I can validate, this was just here to be more explicit with what is being returned on callback. Instead of just having to know the implementation detail of cwrap.

Reply:
Have you checked this? Just to be clear - I'm not sure what `cwrap()` will actually do. I'm just saying that *if* if returns `null` but I don't know if it does.

Reply:
I believe its been updated to return correctly, its been a while since I have had a look but I will double check it.

Reply:
@cameel this has been updated to only handle fallback values which are not undefined. `bindVersion` now does not pass a final fallback which in turn forces the cwrap call to keep in line with the current implementation.